qemu/lib/iozone.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import os
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import postprocess_iozone
+
+
+class IozoneTest(object):
+    """
+    Prepare iozone env, run iozone on guest, clean up iozone env.
+    """
+    def __init__(self, vm, params, test):
+        """
+        run setup of iozone test
+        """
+        self.params = params
+        self.session = vm.wait_for_login()
+        self.test = test
+        self.results_path = ""
+        self.timeout = float(params.get("iozone_timeout", 240))
+
+    def before_run_iozone(self):
+        """
+        Prepare for iozone test
+        """
+        raise NotImplementedError()
+
+    def run_iozone_in_guest(self, disk_index, mountpoint):
+        """
+        Run iozone test in guest
+
+        :param disk_index: index to find disk under test
+        :param mountpoint: mountpoint/disk letter assigned to disk under test
+        :return: iozone run status and result.
+        """
+        iozone_test_cmd_default = "./iozone -azR -r 64k -n 125M -g 512M -i 0 -i 1 %s"
+        iozone_test_cmd = self.params.get("iozone_test_cmd", iozone_test_cmd_default)
+        if self.params.get("os_type") == "windows":
+            if self.params.get("format_disk", "yes") == "yes":
+                logging.info("Format disk")
+                utils_misc.format_windows_disk(self.session, disk_index, mountpoint)
+            test_cmd = iozone_test_cmd % (mountpoint, mountpoint)
+            test_cmd = utils_misc.set_winutils_letter(self.session, test_cmd)
```

Comment:
I'm wondering if this part could be moved to Iozone_Win.before_iozone_run().

Reply:
before_iozone_run()/after_iozone_run() is based on the whole iozone test. ex, when you run iozone with multiple disks, you just need to run only once before_iozone_run() to do some preparation, ex, download iozone test tools. Then you format every test disks and run iozone test on them. Finally, after you finished all iozone tests, you call after_iozone_run() to post results or cleanup iozone. So according to the logic, format_disk and set_winutils_letter should be binded with iozone run for every disk under test.